compiler/p/codegen/OMRRegisterDependency.cpp

```diff
@@ -844,74 +844,86 @@ void TR_PPCRegisterDependencyGroup::assignRegisters(TR::Instruction   *currentIn
       {
       map.addDependency(_dependencies[i], i);
 
-      virtReg = _dependencies[i].getRegister();
       dependentRegNum = _dependencies[i].getRealRegister();
 
       if (dependentRegNum != TR::RealRegister::SpilledReg)
          {
-         if (virtReg->getKind() == TR_GPR)
-            num_gprs++;
-         else if (virtReg->getKind() == TR_FPR)
-            num_fprs++;
-         else if (virtReg->getKind() == TR_VRF)
-            num_vrfs++;
+         virtReg = _dependencies[i].getRegister();
+         switch (virtReg->getKind())
+            {
+            case TR_GPR:
+               ++numGPRs;
+               break;
+            case TR_FPR:
+               ++numFPRs;
+               break;
+            case TR_VRF:
+               ++numVRFs;
+               break;
+            case TR_CCR:
+               if (!virtReg->getAssignedRegister())
```

Comment:
Is this condition enough? In an OOL section if a virtual register is not assigned to a real register it may not always mean the register is spilled. What is this is the first time we are encountering this virtual register in an OOL path but the future use count != total use count. That is, what if the register is used in the mainline hot path but is dead at the merge label of the OOL path? In that case the `virtReg->getAssignedRegister()` will return `NULL` however the virtual register is not spilled. It has just never been assigned.

Small nitpick as well: `getAssignedRegister` does not return a boolean, so shouldn't the if be checking for `virtReg->getAssignedRegister() == NULL` (I really wish we supported `nullptr`).